src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -279,8 +294,7 @@ func (c *controller) updateConfiguringStatusIfNeeded(hosts map[string]inventory_
 	common.SetConfiguringStatusForHosts(c.ic, hosts, logs, false, c.log)
 }
 
-func (c *controller) ApproveCsrs(ctx context.Context, wg *sync.WaitGroup) {
-	defer wg.Done()
+func (c *controller) ApproveCsrs(ctx context.Context) {
```

Comment:
This is exactly the logic of WaitForPredicateWithCancel and it is better to use one function

Reply:
just don't want to change places when it was needed. We can do it in another PR